Inicio do trabalho

In [1]:
import pandas as pd
import seaborn as sb
import matplotlib.pyplot as mtp
import warnings
warnings.filterwarnings('ignore')

Carregando a base de dados "precos_carros_brasil.csv"

In [9]:
dados = pd.read_csv('precos_carros_brasil.csv')

Verificar se existem valores faltantes

In [3]:
dados.isna().any()

year_of_reference     True
month_of_reference    True
fipe_code             True
authentication        True
brand                 True
model                 True
fuel                  True
gear                  True
engine_size           True
year_model            True
avg_price_brl         True
dtype: bool

Tratamento para o caso do campo conter valor faltante

In [19]:
dados = dados.dropna()

In [20]:
dados.isna().sum()

year_of_reference     0
month_of_reference    0
fipe_code             0
authentication        0
brand                 0
model                 0
fuel                  0
gear                  0
engine_size           0
year_model            0
avg_price_brl         0
dtype: int64

Verificando se existem dados duplicados.

In [21]:
dados.duplicated().sum() 

2

Criando duas categorias para separar colunas categoricas e numéricas.

In [11]:
numericas_cols = [col for col in dados.columns if dados[col].dtype != 'object']
categoricas_cols = [col for col in dados.columns if dados[col].dtype == 'object']

In [15]:
dados[categoricas_cols].describe()
dados[numericas_cols].describe()

month_of_reference fipe_code authentication   brand  \
count              202297    202297         202297  202297   
unique                 12      2091         202295       6   
top               January  003281-6   3r6c277cnqcb    Fiat   
freq                24260       425              2   44962   

                                         model      fuel    gear engine_size  
count                                   202297    202297  202297      202297  
unique                                    2112         3       2          29  
top     Palio Week. Adv/Adv TRYON 1.8 mpi Flex  Gasoline  manual         1,6  
freq                                       425    168685  161885       47420

Contagem dos valores por modelo e marca do carro.

In [37]:
print(dados['model'].value_counts())
print(dados['brand'].value_counts())

model
Nada consta.                              65245
Focus 1.6 S/SE/SE Plus Flex 8V/16V 5p       425
Palio Week. Adv/Adv TRYON 1.8 mpi Flex      425
Saveiro 1.6 Mi/ 1.6 Mi Total Flex 8V        400
Focus 2.0 16V/SE/SE Plus Flex 5p Aut.       400
                                          ...  
STEPWAY Zen Flex 1.0 12V Mec.                 2
Saveiro Robust 1.6 Total Flex 16V CD          2
Saveiro Robust 1.6 Total Flex 16V             2
Gol Last Edition 1.0 Flex 12V 5p              2
Polo Track 1.0 Flex 12V 5p                    2
Name: count, Length: 2113, dtype: int64
brand
Nada consta.       65245
Fiat               44962
VW - VolksWagen    44312
GM - Chevrolet     38590
Ford               33151
Renault            29192
Nissan             12090
Name: count, dtype: int64


Dentro os resultados obtidos na Analise Exploratoria, podemos dizer que a base de dados possui onze colunas e a quantidade de linhas é 267542. Dentre os valores armazenados 65246 são valores duplicados e o número de campos sem um valor é de 65245. Para evitar problemas ao processar essa base, os campos sem valores foram tratados, ou seja um valor foi adicionado a estes campos. Podemos também ver que o modelo de carro que aparece com mais frequencia é um empate entre o "Focus 1.6 S/SE/SE Plus Flex 8V/16V 5p" e o "Palio Week. Adv/Adv TRYON 1.8 mpi Flex" que aparecem ambos 425 vezes, ao passo que a marca que mais aparece é a Fiat com 44962 ocorrencias.